### PR TITLE
Enable universal

### PR DESCRIPTION
**Pull request by BenjaminVerble**
- there may be some debatable decisions made here. I've extended app onto the initial state so that child components can "dispatch" / call app.store. This is probably not an ideal way to have stateless components (if that is one of the goals). We should probably figure out a pattern or an api that would allow for dispatcher functions to be defined outside of the view/component tree and then extended onto the state object that goes into virtual-raf. 
- I'm not in a hurry to get this merged. I am most excited about participating here as a learning experience and having a conversation about how best to move this "framework" / pattern forward. I think it will be interesting to see how composability and usability are balanced, for instance. 
- Thanks for your time everybody!

### CODE REVIEW

File: example-server.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/*
+* create the app passing the container element and virtual-dom
+*/
+var app = createApp(vdom)
+
+/*
+* The only way to modify state is to trigger an action
+* the modifer function is where you change state based on the type of an action
+*/
+function modifier (action, state) {
```
Comment (sethvincent) [inline example-server.js:16]: The `modifier` function is almost moot with server-side rendering – at least it seems that way. Are there situations where the modifier function will need to do a lot on the server? I'm not sure. What if we could omit it?

``` js
var render = app.start({
  example: false,
  title: 'My web page title'
})
```

And if the function isn't available just provide `initialState`. I'm not sure if that's weird or not.

Anyway, we could at least simplify this example so `modifier` just returns `state` if it won't really be doing anything.

**Comment by sethvincent** [comment]
This is looking cool!

What if we moved extending `state` with `app` to a separate issue / pull request?

My initial impression is that it shouldn't be necessary, but it would be good to look at some example usage showing how it might work.

**Comment by BenjaminVerble** [comment]
@sethvincent : I was wondering about the modifier function on the server as well. 
**option 1**: reorder modifier and initialState args so that modifier is more easily optional (i.e. not having to check length of arguments passed)
**option 2**: check for # of args. if 1, then just treat arg as initial state (not my favorite option, but leaves api in-tact)
**option 3**: separate server and client versions of virtual-app (same org or same repo, not sure which is better) with shared functions. (I like this option but the direction of the repo changes drastically)

I guess it would be helpful to discuss direction, perhaps in a new issue? Do we have users who need stability? Or can we thrash about with the API until we have what we want? Are we trying to build an ecosystem of composable modules ?(I hope so)

A broader discussion of values might be helpful as well. We seem to have a general consensus of small modules, functional-leaning programming, and a desire to be friendly to many skill-levels? 

I agree that extending app onto state is probably not a good idea. That was probably a sleep-deprivation-induced decision (2 kids at home). This could be done from the consumer's code without much pain. Will leave that out next PR update.

Sorry to expand the scope of the discussion, I'm just enthusiastic about this right now. Happy to continue this outside of this PR comment thread.

**Comment by BenjaminVerble** [comment]
as discussed, I've removed the extending of app onto state. I think improvements to server rendering should be addressed in a separate issue and PR. There are too many different ways we can go on that.

**Comment by yoshuawuyts** [comment]
I don't have any objections to the current state of the PR

**Pull request merged**